client/consensus/grandpa/authorities.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// AuthorityList A list of Grandpa authorities with associated weights.
+// TODO migrate this type and associated functions to this package
+type AuthorityList []types.Authority
+
+// PendingChange A pending change to the authority set.
+//
+// This will be applied when the announcing block is at some depth within
+// the finalized or unfinalized chain.
+type PendingChange struct {
+	nextAuthorities AuthorityList
+	delay           uint
+	canonHeight     uint
+	canonHash       common.Hash
+	delayKind       DelayKind
+}
+
+// EffectiveNumber Returns the effective number this change will be applied at.
+func (pc *PendingChange) EffectiveNumber() uint {
+	return pc.canonHeight + pc.delay
+}
+
+// AuthoritySetChanges Tracks historical authority set changes. We store the block numbers for the last block
+// of each authority set, once they have been finalized. These blocks are guaranteed to
+// have a justification unless they were triggered by a forced change.
+type AuthoritySetChanges []struct {
```

Comment:
you should probably define an `AuthoritySetChange` and define this as `AuthoritySetChanges []AuthoritySetChange`.

Also please use the generic type `N` for block number.

Reply:
I actually do this already in my follow up PR.